builders/interoperability/xcm/xc-registration/foreign-asset-registration.md

```diff
@@ -0,0 +1,134 @@
+
+Deposits are network‑specific and can be adjusted by Moonbeam governance via the `parameters` pallet:
+
+=== "Moonbeam"
+
+     ```text
+     {{ networks.moonbeam.xcm.foreign_asset_deposit.display }} GLMR
+     ```
+
+=== "Moonriver"
+
+     ```text
+     {{ networks.moonriver.xcm.foreign_asset_deposit.display }} MOVR
+     ```
+
+=== "Moonbase Alpha"
+
+     ```text
+     {{ networks.moonbase.xcm.foreign_asset_deposit.display }} DEV
+     ```
+
+## Prerequisites {: #prerequisites }
+
+There are a few prerequisites to be aware of: 
+
+- The sibling parachain's [sovereign account](/builders/interoperability/xcm/core-concepts/sovereign-accounts/){target=\_blank} on Moonbeam must be sufficiently funded to cover the asset deposit and the transaction fees. It's recommended that you have an extra buffer of additional funds for any subsequent transactions. See this [guide to calculating a sovereign account](/builders/interoperability/xcm/core-concepts/sovereign-accounts/){target=\_blank}
+- Your parachain should support XCM V4
+
+## Assemble Your Asset Details {: #assemble-your-asset-details }
+
+Before registering your asset on Moonbeam, you'll need to formulate the following four parameters: asset ID, decimals, symbol, and name. Asset ID can be any integer between 0 and 255 that isn't already used as a foreign asset ID on Moonbeam. 
+
+```typescript
+const ASSET_ID    = 1;
+const DECIMALS    = 18n;
+const SYMBOL      = "TEST";
+const NAME        = "Test Token";
+```
+
+## Construct the Asset MultiLocation {: #construct-the-asset-multilocation }
+
+`assetLocation` is a SCALE‑encoded MultiLocation that pinpoints the existing token on your sibling parachain. There are various ways to define assets and your MultiLocation may including parachain ID, the pallet that manages assets there, and the local asset index. Because the extrinsic executes on Moonbeam, you describe the path from Moonbeam’s perspective: first hop up one level to the Relay ("parents": 1), then down into your parachain (Parachain: <paraId>), the pallet, and the asset index. Moonbeam uses this to verify that the caller actually "contains" the asset before allowing any registration or updates.
+
+Once you've constructed your MultiLocation, keep it handy, as you'll need it in the next step. A typical asset MultiLocation looks like this:
+
+```jsonc
+{
+  "parents": 1,          // up to Relay
+  "interior": {
+    "X3": [              // down to sibling para asset
+      { "Parachain": 4 },
+      { "PalletInstance": 12 },
+      { "GeneralIndex": 15 }  // Arbitrary example values
+    ]
+  }
+}
+```
+
+## Generate the Encoded Call Data {: #generate-the-encoded-call-data }
+
+If you only need the SCALE‑encoded payload—for example, to embed inside an XCM `Transact` dispatched by your runtime—use the snippet below.
+
+```typescript
```

Comment:
While this is good. we are not explaining that they must send an XCM message containing a `Transact` etc etc. You should try the process with XCM Chopsticks so that we provide a bit more detail (without constructing the XCM itself, but just presenting a bit the idea